src/python_inspector/utils_pypi.py

```diff
@@ -1595,11 +1595,32 @@ def fetch_links(
             url, _, _sha256 = anchor_tag["href"].partition("#sha256=")
             if "data-requires-python" in anchor_tag.attrs:
                 python_requires = anchor_tag.attrs["data-requires-python"]
+            url = resolve_relative_url(package_url, url)  # Resolve relative URL
             links.append(Link(url=url, python_requires=python_requires))
         # TODO: keep sha256
         return links
 
 
+def resolve_relative_url(package_url, url):
+    """
+    Resolve a relative URL using the package URL.
```

Comment:
nitpicking: We are using a slightly different style for our docstrings so I would have written something like this
```
"""
Return the resolved `url` URLstring given a `package_url` base URL string
of a package.

For example:
    >>> resolve_relative_url("https://example.com/package", "../path/file.txt")
    'https://example.com/path/file.txt'
"""